### PR TITLE
(maint) Build container from source

### PR DESCRIPTION
**Pull request by underscorgan**
Rather than installing the latest gem from rubygems, do a `gem build`
and install that gem so we're running from source.

### CODE REVIEW

**Comment by underscorgan** [comment]
Running into a couple of things here now.
1. ~`COPY .` doesn't work on windows~
2. ~The way the r10k gemspec is written unless we manually copy in _all_ files from the source directory the gemspec will fail.~

nope, it was me missing the updates to the powershell build script that caused issues.

**Comment by nwolfe** [comment]
@underscorgan Interesting. We've been using `COPY .` in the puppetdb image for a few months now: https://github.com/puppetlabs/puppetdb/blob/master/docker/puppetdb/Dockerfile#L10

We use a `.dockerignore` file in that case though...I wonder if there's something in that file that prevents us from hitting whatever bug you're running into 🤷‍♂

**Comment by underscorgan** [comment]
Oh interesting. I'll investigate this more, thanks @nwolfe !

File: docker/r10k/Dockerfile [hunk 1]
```diff
@@ -1,3 +1,11 @@
+FROM alpine:3.9 as build
+
+RUN apk add --no-cache ruby git
+RUN mkdir /workspace
+WORKDIR /workspace
+COPY . /workspace
```
Comment (Iristyle) [inline docker/r10k/Dockerfile:6]: Based on the Azure fail:

```
 ---> 7827868f48a6
Step 6/23 : RUN gem build r10k.gemspec
 ---> Running in c58a63ca78f4
ERROR:  Gemspec file not found: r10k.gemspec‌
##[error]docker : The command '/bin/sh -c gem build r10k.gemspec' returned a non-zero code: 1
```

I think `cwd` in Azure is not quite the same, so `.` might be a different set of files? Could verify with

```Dockerfile
RUN ls -l /workspace
```

Comment (underscorgan) [reply docker/r10k/Dockerfile:6]: I think this is due to a change in the shared build.ps1, testing

**Pull request merged**